tests/integration.rs

```diff
@@ -32,6 +33,54 @@ pub const TEST_BINS_OUT: &str = "bin";
 pub const TIMEOUT_SECS: u64 = 60 * 60;
 pub const MAX_ASSERT_ELEMENTS: usize = 100;
 
+/// All supported `ENARX_BACKEND` values.
+const BACKENDS: [&str; 4] = ["kvm", "nil", "sev", "sgx"];
+
+/// Returns `true` if `ENARX_BACKEND` is equal to one of the specified `backends`.
+pub fn selected_backend(backends: impl IntoIterator<Item = impl AsRef<str>>) -> bool {
+    var("ENARX_BACKEND")
+        .map(|selected| {
+            backends
+                .into_iter()
+                .find(|backend| selected == backend.as_ref())
+                .is_some()
+        })
+        .unwrap_or_default()
+}
+
+/// Returns `true` if SGX backend should be used by Enarx given the environment
+pub fn is_sgx() -> bool {
+    selected_backend(["sgx"])
+        || cfg!(host_can_test_sgx)
+            && !selected_backend(BACKENDS.iter().filter(|&&backend| backend == "sgx"))
```

Comment:
I don't quite understand the goal with this. `BACKENDS` is const, so we know that this filter always produces an iterator equivalent to `["sgx"]`. So the condition inside the function is saying `selected_backend(["sgx"]) || (foo && !selected_backend(["sgx"]))`. When `selected_backend(["sgx"])` is true, the RHS of the `||` is irrelevant. And when it's false, the `!selected_backend(["sgx"])` is guaranteed to be true, so it has no point. But why would we want to `&&` the `cfg` with the negation of this function in the first place? What is this expressing?

Reply:
Sorry, it's late - that should have been negated.

This is selecting the backend that should the user would expect to be picked by Enarx.
- If `ENARX_BACKEND = "sgx"`, then enarx with use SGX backend or fail trying
- If `/dev/sgx` device is present and no `ENARX_BACKEND` other than `sgx` is selected, then Enarx should pick SGX

Reply:
Although writing this down and looking at this, I think this can be simplified by simply checking if `ENARX_BACKEND` is set at all in the last step, let me do that

Reply:
Original itention was to use this as `selected_backend(["sgx", "sev"])` to assert TEE, for example, but I learned that it would not be as simple and went with functions instead. The previous shape of this helper does not make much sense anymore

Reply:
The new version should make much more sense

Reply:
In fact, I've found an even cleaner solution, see updated version